src/test/fuzz/package_eval.cpp

```diff
@@ -210,50 +210,47 @@ FUZZ_TARGET(ephemeral_package_eval, .init = initialize_tx_pool)
 
     chainstate.SetMempool(&tx_pool);
 
-    LIMITED_WHILE(fuzzed_data_provider.ConsumeBool(), 300)
+    LIMITED_WHILE(fuzzed_data_provider.remaining_bytes() > 0, 300)
     {
         Assert(!mempool_outpoints.empty());
 
         std::vector<CTransactionRef> txs;
 
         // Find something we may want to double-spend with two input single tx
-        std::optional<COutPoint> outpoint_to_rbf{GetChildEvictingPrevout(tx_pool)};
-        bool should_rbf_eph_spend = outpoint_to_rbf && fuzzed_data_provider.ConsumeBool();
+        std::optional<COutPoint> outpoint_to_rbf{fuzzed_data_provider.ConsumeBool() ? GetChildEvictingPrevout(tx_pool) : std::nullopt};
 
         // Make small packages
-        const auto num_txs = should_rbf_eph_spend ? 1 : (size_t) fuzzed_data_provider.ConsumeIntegralInRange<int>(1, 4);
+        const auto num_txs = outpoint_to_rbf ? 1 : (size_t) fuzzed_data_provider.ConsumeIntegralInRange<int>(1, 4);
 
         std::set<COutPoint> package_outpoints;
         while (txs.size() < num_txs) {
-
-            // Last transaction in a package needs to be a child of parents to get further in validation
-            // so the last transaction to be generated(in a >1 package) must spend all package-made outputs
-            // Note that this test currently only spends package outputs in last transaction.
-            bool last_tx = num_txs > 1 && txs.size() == num_txs - 1;
-
             // Create transaction to add to the mempool
-            const CTransactionRef tx = [&] {
+            txs.push_back([&] {
                 CMutableTransaction tx_mut;
                 tx_mut.version = CTransaction::CURRENT_VERSION;
                 tx_mut.nLockTime = 0;
-                // Last tx will sweep half or more of all outpoints from package
-                const auto num_in = should_rbf_eph_spend ? 2 :
+                // Last transaction in a package needs to be a child of parents to get further in validation
+                // so the last transaction to be generated(in a >1 package) must spend all package-made outputs
+                // Note that this test currently only spends package outputs in last transaction.
+                bool last_tx = num_txs > 1 && txs.size() == num_txs - 1;
+                const auto num_in = outpoint_to_rbf ? 2 :
                     last_tx ? fuzzed_data_provider.ConsumeIntegralInRange<int>(package_outpoints.size()/2 + 1, package_outpoints.size()) :
                     fuzzed_data_provider.ConsumeIntegralInRange<int>(1, 4);
-                auto num_out = should_rbf_eph_spend ? 1 : fuzzed_data_provider.ConsumeIntegralInRange<int>(1, 4);
+                auto num_out = outpoint_to_rbf ? 1 : fuzzed_data_provider.ConsumeIntegralInRange<int>(1, 4);
```

Comment:
nit: Could add `const` to `num_out` for symmetry with `num_in` as you are touching this line (pointed out in https://github.com/bitcoin/bitcoin/pull/30239#discussion_r1832788903).

Reply:
done